addons/mail/models/ir_attachment.py

```diff
@@ -57,5 +63,60 @@ def _attachment_format(self, commands=False):
                     'res_id': attachment.res_id,
                     'res_model': attachment.res_model,
                 })
-            res_list.append(res)
-        return res_list
+            attachments.append(res)
+        return attachments
+
+    @api.model
+    def _get_attachment_data_from_url(self, url):
+        """
+        This will create attachment data based on what we can read from the URL.
+        If the URL is an HTML page, this will parse the OpenGraph meta data.
+        If the URL is an image, this will create an image attachment.
+        """
+        data = {}
+        request_image = None
+        # data = base64.b64encode(data)
+        try:
+            page = requests.get(url, timeout=10)
+        except requests.exceptions.RequestException:
+            return False
+
+        if page.status_code != requests.codes.ok:
+            return False
+        image_mimetype = [
+            'image/bmp',
+            'image/gif',
+            'image/jpeg',
+            'image/png',
+            'image/tiff',
+            'image/x-icon',
+        ]
+        if page.headers['content-type'] in image_mimetype:
+            image = base64.b64encode(page.content)
+            image = image_process(image, size=(300,300), verify_resolution=True)
+            data = {
+                'url': url,
+                'name': url,
+                'description': False,
+                'datas': image,
+                'mimetype': page.headers['content-type'],
+            }
+            return data
+        elif 'text/html' in page.headers['content-type']:
+            tree = html.fromstring(page.content)
+            if tree.xpath('//meta[@property="og:title"]/@content'):
+                image = None
+                if tree.xpath('//meta[@property="og:image"]/@content'):
+                    image = tree.xpath('//meta[@property="og:image"]/@content')[0]
+                    request_image = requests.get(image, timeout=10)
+                    image = base64.b64encode(request_image.content)
+                    image = image_process(image, size=(300,300), verify_resolution=True)
+                data = {
+                    'url': url,
+                    'name': tree.xpath('//meta[@property="og:title"]/@content')[0] if tree.xpath('//meta[@property="og:title"]/@content') else url,
+                    'datas': image if image else False,
+                    'description': tree.xpath('//meta[@property="og:description"]/@content')[0] if tree.xpath('//meta[@property="og:description"]/@content') else False,
```

Comment:
kind of hard to read (and not performant) to repeat each xpath twice, it might be better to store its result on a variable